eosiojava/src/main/java/one/block/eosiojava/utilities/EOSFormatter.java

```diff
@@ -818,27 +818,45 @@ public static String extractSerializedTransactionFromSignable(@NotNull String eo
      * <p>
      * Signable signature structure:
      * <p>
-     * chainId + serialized transaction + 32 bytes of 0
+     * chainId + serialized transaction + 32 bytes of serializedContextFreeData
      *
      * @param serializedTransaction - the serialized transaction to be converted to signable transaction
      * @param chainId - the chain id will be used inside the signature transaction structure.
+     * @param serializedContextFreeData - the serialized context free data to be converted with transaction
      * @return - Signable transaction
      * @throws EOSFormatterError if inputs are invalid
      */
     public static String prepareSerializedTransactionForSigning(@NotNull String serializedTransaction,
-            @NotNull String chainId) throws EOSFormatterError {
-        if (serializedTransaction.isEmpty() || chainId.isEmpty()) {
+            @NotNull String chainId, @NotNull String serializedContextFreeData) throws EOSFormatterError {
+        if (serializedTransaction.isEmpty() || chainId.isEmpty() || serializedContextFreeData.isEmpty()) {
```

Comment:
Current implementations that do not pass context free data through now have an overload to always pass `Hex.toHexString(new byte[32])`: https://github.com/EOSIO/eosio-java/pull/115/files#diff-f2d1228325d78f0d7bfe71d2f26ec499R855

If someone attempts to use this method signature (https://github.com/EOSIO/eosio-java/pull/115/files#diff-f2d1228325d78f0d7bfe71d2f26ec499R829) with an empty string, it will throw this error because it's required and always has been: https://github.com/EOSIO/eosio-java/pull/115/files#diff-f2d1228325d78f0d7bfe71d2f26ec499L834

Reply:
Cool. Thanks for the explanation. We should put this warning into our READE me warn user about this method.